tests/dicts/keypath/test_keypath_dict_list_wildcard.py

```diff
@@ -0,0 +1,502 @@
+
+        # wrong index => default value should be used
+        self.assertTrue(b.get("*.b[2].*", 2), 2)
+        self.assertTrue(b["*.b[1][*]"], [4, 5, 6, [0]])
+
+        with self.assertRaises(KeyError):
+            self.assertFalse(b["*.a"], None)
+        with self.assertRaises(KeyError):
+            self.assertFalse(b["*.*"], None)
+
+        self.assertEqual(b.get("*.*", "not-existing"), "not-existing")
+
+    def test_complex_wildcard_usage(self):
+        d = {
+            "a": {
+                "b": [
+                    {
+                        "c": [
+                            {"x": 10, "y": 20},
+                            {"x": 11, "y": 21},
+                            {"x": 12, "y": 22},
+                        ],
+                    },
+                    {
+                        "c": [
+                            {"x": 20, "y": 30},
+                            {"x": 21, "y": 31},
+                            {"x": 22, "y": 32},
+                        ],
+                    },
+                ],
+            },
+        }
+        b = KeypathDict(d)
+        self.assertEqual(
+            b.get("a.b[0]"),
+            {
+                "c": [
+                    {"x": 10, "y": 20},
+                    {"x": 11, "y": 21},
+                    {"x": 12, "y": 22},
+                ],
+            },
+        )
+        self.assertEqual(
+            b.get("a.b[*]"),
+            [
+                {
+                    "c": [
+                        {"x": 10, "y": 20},
+                        {"x": 11, "y": 21},
+                        {"x": 12, "y": 22},
+                    ],
+                },
+                {
+                    "c": [
+                        {"x": 20, "y": 30},
+                        {"x": 21, "y": 31},
+                        {"x": 22, "y": 32},
+                    ],
+                },
+            ],
+        )
+        self.assertEqual(
+            b.get("a.b[0].c"),
+            [
+                {"x": 10, "y": 20},
+                {"x": 11, "y": 21},
+                {"x": 12, "y": 22},
+            ],
+        )
+        self.assertEqual(
+            b.get("a.b[1].c"),
+            [
+                {"x": 20, "y": 30},
+                {"x": 21, "y": 31},
+                {"x": 22, "y": 32},
+            ],
+        )
+        self.assertEqual(b.get("a.b[1].c[1]"), {"x": 21, "y": 31})
+        self.assertEqual(
+            b.get("a.b[*].c"),
+            [
+                [{"x": 10, "y": 20}, {"x": 11, "y": 21}, {"x": 12, "y": 22}],
+                [{"x": 20, "y": 30}, {"x": 21, "y": 31}, {"x": 22, "y": 32}],
+            ],
+        )
+        self.assertEqual(
+            b.get("a.b[*].c[*].x"),
+            [10, 11, 12, 20, 21, 22],
+        )
+        self.assertEqual(
+            b.get("a.b[*].c[*].x[-1]"),
+            22,
+        )
+
+    def test_complex_wildcard_non_consecutive(self):
+        d = {
+            "a": [
+                {
```

Comment:
Should add more than one item to each list to verify that the final value is chained correctly at all levels.